src/test/java/integration/realprojects/ApacheCommonsCliIntegrationTest.java

```diff
@@ -44,28 +44,28 @@ public void isInnerClassRefactoring(){
 	public void isInnerClassStable(){
 		List<StableCommit> areInnerClassesInStable = getStableCommits().stream().filter(commit ->
 				commit.getClassMetrics().isInnerClass()).collect(Collectors.toList());
-		Assert.assertEquals(43, areInnerClassesInStable.size());
+		Assert.assertEquals(65, areInnerClassesInStable.size());
 
 		List<StableCommit> areNotInnerClassesInStable = getStableCommits().stream().filter(commit ->
 				!commit.getClassMetrics().isInnerClass()).collect(Collectors.toList());
-		Assert.assertEquals(1702, areNotInnerClassesInStable.size());
+		Assert.assertEquals(2195, areNotInnerClassesInStable.size());
 
-		Assert.assertEquals(1745, getStableCommits().size());
+		Assert.assertEquals(2260, getStableCommits().size());
```

Comment:
what do these numbers change? they shouldn't change, right?

Reply:
We are now collecting more stable instances because we collect a stable instance once for each applicable level, e.g. if an instance has methods with variables, we now collect instances for the class level, the method level, and the variable level. This creates lots of duplicate entries in the DB, but simplifies the usage and understanding a lot.